Decisions.AdobeSign/Steps/AbstractStep.cs

```diff
@@ -69,18 +69,15 @@ private AdobeSignConnection CreateConnection(string id)
         }
         private string GetBaseApi(OAuthToken oAuthToken)
         {
-            string authorizationResponse = oAuthToken.FullAuthorizationResponse;
-            if(String.IsNullOrEmpty(authorizationResponse))
-                throw new LoggedException("Can't extract AdobeSign's base URL");
-            int qIdx = authorizationResponse.IndexOf('?');
-            if (qIdx == -1)
-                throw new LoggedException("Can't extract AdobeSign's base URL");
-            string queryString = authorizationResponse.Substring(qIdx + 1);
-            NameValueCollection qs = HttpUtility.ParseQueryString(queryString);
-
-            var endpoint = qs["api_access_point"];
-            if (qs == null) throw new LoggedException("Can't extract AdobeSign's base URL");
-            return endpoint;
+            try
+            {
+                JObject keyValuePairs = JsonConvert.DeserializeObject(oAuthToken.FullAccessTokenResponse) as JObject;
+                return keyValuePairs?["api_access_point"]?.ToObject<string>();
```

Comment:
The ?s here are going to swallow the null exception and return it as a value. 
Should it fail and hit the exception if the response format is not correct, instead?

Reply:
keyValuePairs will always be non-null from my experience with JsonConvert, may just end up being an empty list though.

as for the missing key after the bracket, there is an argument to be made that this might be better captured beforehand.  you are right there I suppose. 

I am going to capture the null value case and throw an exception instead of immediately returning, good call

Reply:
updated

Reply:
If you update to address a comment please mark it as resolved.

Reply:
literally was doing just that as you replied hah